core/src/include/ModelArray.hpp

```diff
@@ -272,15 +274,19 @@ class ModelArray {
      * @param dim The dimension to be altered.
      * @param length The new length of the dimension.
      */
-    static void setDimension(Dimension dim, size_t length);
+#ifdef USE_MPI
+    static void setDimension(Dimension dim, size_t global_length, size_t local_length, size_t size);
+#else
+    static void setDimension(Dimension dim, size_t global_length);
+#endif
```

Comment:
I did think about this, but it's much nicer to get compile time warning (in my opinion). Mostly because it stung me a few times already :sweat:  and I think it could catch future devs out as well.

Happy to discuss in dev meeting though as I don't have a strong opinion either way. I did initially code as optional but I removed that when it prevented me finding bugs.

Reply:
tldr; using the optional approach, it is possible to use the wrong function call and you will only find out at run time.